From 1483de4532e9de6fe8d904a361f20a2ba1dbbc07 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 23 Nov 2022 16:08:17 +0200
Subject: [PATCH] [SW-0] gaudi3: fix debug straps numbers

debug straps numbers are updated to match the gaudi3 gpio spec

Change-Id: I06dd974042e831ae48c70895a4c5404f6aad6be4
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index e7afce1..cb53567 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -275,16 +275,15 @@
 	u32 boot_strap = regmap_readl(mmD0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L);
 	u32 strap_mask;
 
-	/* TODO: verify relevant straps for debug functions */
 	switch (func) {
 	case HL_DEBUG_HW_PCIE:
-		strap_mask = BIT(12); /* Dump Select[11] */
+		strap_mask = BIT(5); /* BTL_DISABLE */
 		break;
 	case HL_DEBUG_HW_DRAM:
-		strap_mask = BIT(11); /* Dump Select[10] */
+		strap_mask = BIT(4); /* RSVD[1] */
 		break;
 	case HL_DEBUG_HW_I2C:
-		strap_mask = BIT(10); /* Dump Select[9] */
+		strap_mask = BIT(2); /* RSVD[0] */
 		break;
 	default:
 		preboot_puts_hex("Unknown H/W function ", func, false);
